src/Raven.Studio/typescript/viewmodels/database/settings/databaseSettings.ts

```diff
@@ -0,0 +1,444 @@
+
+        saveSettingsModel.result.done(result => {
+            if (result.can) {
+                this.spinners.save(true);
+
+                new saveDatabaseSettingsCommand(this.activeDatabase(), settingsToSaveSorted)
+                    .execute()
+                    .done(() => {
+                        this.fetchDatabaseSettings(this.activeDatabase())
+                            .done(() => this.processAfterFetch());
+                    })
+                    .always(() => this.spinners.save(false))
+                }
+            });
+
+        app.showBootstrapDialog(saveSettingsModel);
+    }
+
+    refresh() {
+        this.confirmRefresh()
+            .then(result => {
+                if (result.can) {
+                    this.spinners.refresh(true);
+
+                    this.fetchDatabaseSettings(this.activeDatabase(), true)
+                        .done(() => {
+                            this.processAfterFetch();
+                            this.dirtyFlag().reset();
+                        })
+                        .always(() => this.spinners.refresh(false));
+                }
+            });
+    }
+
+    private processAfterFetch() {
+        if (this.viewMode() === "editMode") {
+            this.setupEditMode();
+        } else {
+            this.computeEntriesToShow();
+        }
+    }
+    
+    confirmRefresh() {
+        let text = `<div>Clicking Refresh will reload all database settings from the server.</div>`;
+        if (this.dirtyFlag().isDirty()) {
+            text = "<div><span class='bg-warning text-warning'><i class='icon-warning margin-right margin-right-sm'></i>You have unsaved changes !</span></div>" + text;
+        }
+
+        return this.confirmationMessage("Are you sure?", text,
+            {
+                html: true,
+                buttons: ["Cancel", "Refresh"]
+            });
+    }
+    
+    setGroupToShow(groupType: entriesGroupType) {
+        this.groupToShow(groupType);
+    }
+
+    exitEditMode() {
+        this.viewMode("summaryMode");
+        this.computeEntriesToShow();
+    }
+    
+    switchToEditMode() {
+        if (this.editModeHasBeenEntered) {
+            this.viewMode("editMode");
+            this.setupEditMode();
+        } else {
+            this.confirmationMessage("Are you an expert?",
+                "Modify the database settings only if you know what you are doing!", { buttons: ["Cancel", "OK"] })
+                .done(() => {
+                    this.editModeHasBeenEntered = true;
+                    this.viewMode("editMode");
+                    this.initCategoryGrid();
+                    this.setupEditMode(false);
+                });
+        }
+    }
+    
+    private setupEditMode(reset: boolean = true) {
+        if (reset) {
+            this.categoriesGridController().reset(false);
+        }
+        
+        this.setCategory(this.selectedCategory() || this.allEntries()[0].data.Metadata.Category);
```

Comment:
don't we have computed with category list?

Reply:
done